### DISCUSSION TITLE
Query about multilevel keys which don't exist already

### DISCUSSION THREAD

**Discussion by sreerajchundayil** [nlohmann/json#4188] [Q&A] [answered]
using json=nlohmann::json ;

 1)   Is this code safe from any kind of crash if key "A" and "B" doesn't exist already?
  ```
 json js;
 json &myRef = js["A"]["B"]["C"];
```
  

2)
```
  json& test = js["A"]["B"]["C"];
  test["D"] = 1;
  cout << test.dump(2) << "\n";

```
Why does this code just prints 

> {
>   "D": 1
> }

Skipping all "A" and "B" ?

**Comment by gregmarr** [comment, marked as answer]
1. Yes
2. Because you dumped test and not js.
